packages/experiments/src/components/KeytipLayer/KeytipLayer.tsx

```diff
@@ -0,0 +1,248 @@
+import * as React from 'react';
+import { IKeytipLayerProps } from './KeytipLayer.types';
+import { Keytip, IKeytipProps, KeytipTransitionModifier } from '../Keytip';
+import {
+  autobind,
+  BaseComponent
+} from '../../Utilities';
+import { Layer } from 'office-ui-fabric-react/lib/Layer';
+import { KeyCodes, findIndex } from '../../Utilities';
+import { convertSequencesToKeytipID, fullKeySequencesAreEqual } from '../../utilities/keysequence/IKeySequence';
+import { IKeytipTransitionKey } from '../../utilities/keysequence/IKeytipTransitionKey';
+import { KeytipManager } from '../../utilities/keytip/KeytipManager';
+import { ktpFullPrefix, ktpSeparator } from '../../utilities/keytip/KeytipUtils';
+
+export interface IKeytipLayerState {
+  inKeytipMode: boolean;
+  keytips: IKeytipProps[];
+}
+
+const defaultStartSequence: IKeytipTransitionKey = {
+  key: 'Meta', modifierKeys: [KeytipTransitionModifier.alt]
+};
+
+const defaultExitSequence: IKeytipTransitionKey = {
+  key: 'Meta', modifierKeys: [KeytipTransitionModifier.alt]
+};
+
+const defaultReturnSequence: IKeytipTransitionKey = {
+  key: 'Escape'
+};
+
+/**
+ * A layer that holds all keytip items
+ *
+ * @export
+ * @class KeytipLayer
+ * @extends {BaseComponent<IKeytipLayerProps>}
+ */
+export class KeytipLayer extends BaseComponent<IKeytipLayerProps, IKeytipLayerState> {
+  public static defaultProps: IKeytipLayerProps = {
+    keytipStartSequences: [defaultStartSequence],
+    keytipExitSequences: [defaultExitSequence],
+    keytipReturnSequences: [defaultReturnSequence],
+    id: ktpFullPrefix + 'Alt' + ktpSeparator + 'Meta'
+  };
+
+  private _keytipManager: KeytipManager = KeytipManager.getInstance();
+
+  // tslint:disable-next-line:no-any
+  constructor(props: IKeytipLayerProps, context: any) {
+    super(props, context);
+
+    this.state = {
+      inKeytipMode: false,
+      keytips: []
+    };
+
+    this._keytipManager.init(this);
+  }
+
+  /**
+   * Register a keytip in this layer
+   *
+   * @param keytipProps - IKeytipProps to add to this layer
+   */
+  public registerKeytip(keytipProps: IKeytipProps): void {
```

Comment:
Rather than an API which takes a single keytip, why not take in a map of keytips?

Reply:
register should only be called from one component''s render() at a time, so I don't see a use case for registering multiple right now but if we come across one we can definitely do it! would be very easy